### DISCUSSION TITLE
Asking for help to understand a '{} item' in a string

### DISCUSSION THREAD

**Discussion by rffontenelle** [rpm-software-management/dnf5#1160] [General]
`{} item` in the string below results in what content in the built dnf5? How is it used?

https://github.com/rpm-software-management/dnf5/blob/91114403b68a7f45e5fc455d79bb0051f6a66cbf/dnf5-plugins/copr_plugin/copr_repo.cpp#L375-L386

**Comment by ppisar** [comment]
libdnf5::utils::sformat() returns a string where `{}` sequences in the input format string are replaced with a string representation of arguments following the format string. It's a tiny wrapper for fmt library. A syntax of the format string is defined at <https://fmt.dev/latest/syntax.html>.

E.g. if `n=5` and `string="foo:bar"`, then the resulting string saved in `msg` variable will be `Can't find 5 item in foo:bar`. This example demonstrates that the message is not a perfect English. Perfect English would read `Can't find the 5th item in foo:bar`. And `2nd item` for `n=2` etc. One could fix it with gettext plural forms to cover all possible numbers.

**Reply by rffontenelle** [reply]
Ordinal number doesn't seem to me as a plural-form scenario. As far as I know, Gettext doesn't support making ordinal out of number (e.g. "1st" from number "1")

How about making it `item {}` (item 1, item 2, etc.) instead of thr current 1 item (for first), 2 item (for second), etc. ?

**Reply by ppisar** [reply]
You are right that ordinal numbers in English have a "flection" independent from cardinal numbers. Gettext supports only a single pluralization rule per catalog and  that is usually used for cardinal numbers. Placing the number into a postposition is probably the best approach here.

**Reply by rffontenelle** [reply]
Thanks! I filed the PR #1199 to make it `item {}`.
